### DISCUSSION TITLE
A question about determining the Power Spectral Density estimate

### DISCUSSION THREAD

**Discussion by Vidhyarth99** [groops-devs/groops#281] [Q&A] [answered]
Hello,

I have a data derived from GRACE mission with a sampling interval of 2 seconds. The data has missing samples at several instances, with which I need to determine the Power Spectral Density estimate. So, is there any function in GROOPS that can be used to conduct spectral analysis and arrive at the PSD for a dataset that has missing samples?

Best,
Vidhyarth T S

**Comment by tmayerguerr** [comment, marked as answer]
Hi Vidhyarth,

Maybe [Instrument2PowerSpectralDensity](https://groops-devs.github.io/groops/html/Instrument2PowerSpectralDensity.html) is the program you are looking for.

regards
Torsten

**Reply by Vidhyarth99** [reply]
Hello Torsten,

Thank you for your response.

Regarding the function you mentioned, should the first column represent time in seconds (only for the available data)? Additionally, I would like to know if I can use CovarianceFunction2PowerSpectralDensity to obtain the power spectral density (PSD) from a covariance matrix (with time in the first column followed by the covariance matrix) or is it different?

Best,
Vidhyarth T S
